packages/mui-material/src/MobileStepper/MobileStepper.js

```diff
@@ -115,9 +115,11 @@ const MobileStepper = React.forwardRef(function MobileStepper(inProps, ref) {
     ...other
   } = props;
 
+  const active = activeStep >= steps ? steps - 1 : activeStep;
```

Comment:
I think active step will never be greater than steps.

Reply:
@ZeeshanTamboli we added that check in the scenario where the total number of steps gets changed to a value lower than the current step.

Reply:
Can you add a test case for that?

Reply:
The test case has been added @ZeeshanTamboli